diff --git a/compile/ninja/macos.ninja b/compile/ninja/macos.ninja
index df09dcd..0c8ef80 100644
--- a/3rd/luamake/compile/ninja/macos.ninja
+++ b/3rd/luamake/compile/ninja/macos.ninja
@@ -3,7 +3,7 @@ bin = $builddir/bin
 obj = $builddir/obj
 rule C_source_lua
   command = gcc -MMD -MT $out -MF $out.d -std=c11 -O2 -Wall $
-      -DLUA_USE_MACOSX -DNDEBUG -mmacosx-version-min=10.13 -o $out -c $in
+      -DLUA_USE_MACOSX -DNDEBUG -mmacosx-version-min=10.12 -o $out -c $in
   description = Compile C   $out
   depfile = $out.d
   deps = gcc
@@ -47,7 +47,7 @@ build $obj/source_lua/lzio.obj: C_source_lua 3rd/bee.lua/3rd/lua/lzio.c
 rule C_lua
   command = gcc -MMD -MT $out -MF $out.d -std=c11 -O2 -Wall $
       -fvisibility=hidden -DLUA_USE_MACOSX -DNDEBUG -I3rd/bee.lua $
-      -mmacosx-version-min=10.13 -o $out -c $in
+      -mmacosx-version-min=10.12 -o $out -c $in
   description = Compile C   $out
   depfile = $out.d
   deps = gcc
@@ -85,7 +85,7 @@ rule CXX_bee
   command = g++ -MMD -MT $out -MF $out.d -std=c++17 -O2 -Wall $
       -fvisibility=hidden -I3rd/bee.lua/3rd/lua -I3rd/bee.lua/3rd/lua-seri $
       -I3rd/bee.lua/bee/nonstd -I3rd/bee.lua -DNDEBUG -fPIC $
-      -mmacosx-version-min=10.13 -o $out -c $in
+      -mmacosx-version-min=10.12 -o $out -c $in
   description = Compile C++ $out
   depfile = $out.d
   deps = gcc
@@ -131,7 +131,7 @@ build $bin/bee.so: LINK_bee $obj/bee/lua-seri.obj $obj/bee/error.obj $
 rule CXX_bootstrap
   command = g++ -MMD -MT $out -MF $out.d -std=c++17 -O2 -Wall $
       -fvisibility=hidden -I3rd/bee.lua/3rd/lua -DLUA_USE_MACOSX -DNDEBUG $
-      -I3rd/bee.lua -mmacosx-version-min=10.13 -o $out -c $in
+      -I3rd/bee.lua -mmacosx-version-min=10.12 -o $out -c $in
   description = Compile C++ $out
   depfile = $out.d
   deps = gcc
diff --git a/bee/fsevent/fsevent_osx.cpp b/bee/fsevent/fsevent_osx.cpp
index fe3bcd7..f1cbe8b 100644
--- a/3rd/luamake/3rd/bee.lua/bee/fsevent/fsevent_osx.cpp
+++ b/3rd/luamake/3rd/bee.lua/bee/fsevent/fsevent_osx.cpp
@@ -1,4 +1,5 @@
 #include <bee/fsevent/fsevent_osx.h>
+#include <functional>
 #include <fmt/format.h>
 #include <thread>
